core/controllers/acl_decorators.py

```diff
@@ -1997,6 +1997,47 @@ def test_can_modify(self, exploration_id, **kwargs):
     return test_can_modify
 
 
+def can_perform_tasks_in_taskqueue(handler):
+    """Decorator to ensure that the handler is being called by task scheduler or
+    by a superadmin of the application.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also ensures that
+        the handler can only be executed if it is called by task scheduler or by
+        a superadmin of the application.
+    """
+
+    def test_can_perform(self, **kwargs):
+        """Checks if the handler is called by task scheduler or by a superadmin
+        of the application.
+
+        Args:
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            UnauthorizedUserException. The user does not have
+                credentials to access the page.
+        """
+        # The X-AppEngine-QueueName header is set inside AppEngine and if
+        # a request from outside comes with this header AppEngine will get
+        # rid of it.
```

Comment:
This has been tested manually? (Want to check if it still applies in Python 3 -- maybe add a link in the comment to docs confirming that.)

Reply:
Yes, this was tested and I added the link.